### ISSUE TITLE
Reassigning `type` fails to typecheck

### ISSUE THREAD

**Issue by jolaf** [python/mypy#17594]
The following code:
```python3
_type = type
v: _type[object]
```
produces the following output:
```shell
$ mypy --enable-incomplete-feature=NewGenericSyntax test.pyi 
test.pyi:3:4: error: "type" expects no type arguments, but 1 given  [type-arg]
    v: _type[object]
       ^
Found 1 error in 1 file (checked 1 source file)
```
This looks like a false positive to me.
```shell
$ mypy --version
mypy 1.11.0 (compiled: yes)
$ python3 --version
Python 3.12.3
$ lsb_release -d
No LSB modules are available.
Description:	Ubuntu 24.04 LTS
```

**Comment by brianschubert** [comment]
This works if you include explicit type parameters when creating the type alias:
```python
from typing_extensions import reveal_type

T = TypeVar("T")

_type = type[T]
v: _type[object]

reveal_type(_type)  # N: Revealed type is "typing._SpecialForm"
reveal_type(v)      # N: Revealed type is "type[builtins.object]"
```
I'm not sure how well defined it is to create generic type aliases without including explicit type parameters. Without type parameters, `_type` looks like a [higher kinded type](https://en.wikipedia.org/wiki/Kind_(type_theory)), which AFAIK aren't supported in Python's type system.

**Comment by jolaf** [comment]
Unfortunately, this won't help.

The idea was to use another word for `type` operations as the word `type` has to be used for other purposes in the particular piece of code.

**Comment by erictraut** [comment]
> I'm not sure how well defined it is to create generic type aliases without including explicit type parameters

This behavior is not well specified in the typing spec for [old-style type aliases](https://typing.readthedocs.io/en/latest/spec/aliases.html#type-aliases) (PEP 484), and you'll see different behaviors across type checkers. For example, pyright treats this case differently than mypy and does not generate an error for the above code.

The behavior *is* well-defined if you define the type alias using a [TypeAlias](https://typing.readthedocs.io/en/latest/spec/aliases.html#typealias) annotation (PEP 613) or a [type statement](https://typing.readthedocs.io/en/latest/spec/aliases.html#type-statement) (PEP 695). If you are using Python 3.10 or newer, I recommend moving away from PEP 484 type aliases.

Here's a workaround that avoids a type alias definition altogether. It works with all type checkers including mypy.

```python
from builtins import type as _type

v: _type[object]
```
